### PR TITLE
feat(argo-workflow): Add pathType value to ingress

### PR DESCRIPTION
**Pull request by thomascjohnson**
Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md#versioning)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo).
* [x] My build is green ([troubleshooting builds](https://argoproj.github.io/argo-cd/developer-guide/ci/)).

Changes are automatically published when merged to `master`. They are not published on branches.

### CODE REVIEW

**Comment by thomascjohnson** [comment]
Hi folks,

I am running into the issue on GKE that inspired this [PR](https://github.com/argoproj/argo-helm/pull/822/files). I'm not sure if my comits count as signed, though, because I did all of this quickly through the Github UI. If not, let me know.

Thanks!

File: charts/argo-workflows/Chart.yaml [hunk 1]
```diff
@@ -16,3 +16,4 @@ maintainers:
 annotations:
   artifacthub.io/changes: |
     - "[Changed]: Argo workflows values file no longer has examples referencing ArgoCD"
```
Comment (mkilchhofer) [inline charts/argo-workflows/Chart.yaml:18]: The old changes need to be removed
```suggestion
```

File: charts/argo-workflows/templates/server/server-ingress.yaml [hunk 2]
```diff
@@ -36,7 +36,7 @@ spec:
           {{- range $p := $paths }}
           - path: {{ $p }}
             {{- if eq (include "argo-workflows.ingress.apiVersion" $) "networking.k8s.io/v1" }}
-            pathType: Prefix
+            pathType: {{ $pathType }}
```
Comment (mkilchhofer) [inline charts/argo-workflows/templates/server/server-ingress.yaml:39]: You also need to add this on top of this file (eg. line 6):
```
{{- $pathType := .Values.server.ingress.pathType -}}
```
Ref: https://github.com/argoproj/argo-helm/pull/822/files#diff-cee8e938567dd2c92bf1ac995920fc961d20e4cf5a18b6220fbf869d6bff93e0R6

**Review by mkilchhofer — CHANGES_REQUESTED** [review]
See comments

File: charts/argo-workflows/templates/server/server-ingress.yaml (hunk 2 above)
Comment (thomascjohnson) [reply charts/argo-workflows/templates/server/server-ingress.yaml:40]: Thank you for catching this!

**Comment by thomascjohnson** [comment]
@mkilchhofer Thank you very much for the helpful comments. Would you mind having a look now?

**Review by mkilchhofer — APPROVED** [review]

**Pull request merged**